azure-pipelines.yml

```diff
@@ -70,20 +70,20 @@ jobs:
   ################################################################################
   # azure-pipelines-task-lib
   ################################################################################
-  
+
   # npm install
   - task: Npm@1
     displayName: (azure-pipelines-task-lib) npm install
     inputs:
       command: install
       workingDir: node
 
-  # use node 10
+  # use node 16
   - task: NodeTool@0
-    displayName: (azure-pipelines-task-lib) use node 10.23.0
+    displayName: (azure-pipelines-task-lib) use node 16.13.0
     inputs:
-      versionSpec: "10.23.0"
-      
+      versionSpec: "16.13.0"
+
   # build/test
```

Comment:
Could you please add a `PublishPipelineArtifact` step to publish the SDK build as a pipeline artifact?

Also, I believe we can extract common pipeline steps to the template.